src/mapgen/mapgen.h

```diff
@@ -133,20 +133,20 @@ struct MapgenParams {
 
 	BiomeParams *bparams = nullptr;
 
-	s16 mapgen_edge_min = -MAX_MAP_GENERATION_LIMIT;
-	s16 mapgen_edge_max = MAX_MAP_GENERATION_LIMIT;
-
 	virtual void readParams(const Settings *settings);
 	virtual void writeParams(Settings *settings) const;
 
 	bool saoPosOverLimit(const v3f &p);
 	s32 getSpawnRangeMax();
 
 private:
-	void calcMapgenEdges();
+	void calcMapgenEdges(s16 mg_limit, s16 csize_n, s16 ccmin, s16 ccmax,
+	s16 ccfmin, s16 ccfmax, s16 *mapgen_edge_min, s16 *mapgen_edge_max);
+	void calcMgEdgesAndSpawnLimit();
 
 	float m_sao_limit_min = -MAX_MAP_GENERATION_LIMIT * BS;
 	float m_sao_limit_max = MAX_MAP_GENERATION_LIMIT * BS;
+	s32 m_spawn_range_max = 4000;
```

Comment:
`MAX_MAP_GENERATION_LIMIT`

Reply:
I'll explain with a comment, it's equal to the maximum spawn search range in 'find spawn pos'.

Reply:
Done.